introduce scs-0103-v1 v1.1 (backwards compatible, compliance-preserving) (#677)
SovereignCloudStack/standards: Standards/scs-0103-v1-standard-flavors.md

## Introduction

This is v1.1 of the standard, which lifts the following restriction regarding the property `scs:name-vN`:
this property may now be used on any flavor, rather than standard flavors only. In addition, the "vN" is
now interpreted as "name variant N" instead of "version N of the naming standard". Note that this change
indeed preserves compliance, i.e., compliance with v1.0 implies compliance with v1.1.

## Terminology

extra_specs
  Additional properties on an OpenStack flavor, see

>   https://docs.openstack.org/nova/2024.1/user/flavors.html#extra-specs
>   and
>   https://docs.openstack.org/nova/2024.1/configuration/extra-specs.html

Reviewer: ```suggestion
  <https://docs.openstack.org/nova/2024.1/user/flavors.html#extra-specs>
  and
  <https://docs.openstack.org/nova/2024.1/configuration/extra-specs.html>
```

should suffice to please the markdown linter gods

Author: I think this linter is outdated. The newer Docusaurus prefers bare links.

Author: @maxwolfs can you please comment on this?

Author: I have removed the bare links by adding descriptions.

Whenever ANY of these are present on ANY flavor, the corresponding semantics must be satisfied.

> The property `scs:name-vN` is to be interpreted as "name variant N". This name scheme is designed to be

Reviewer: ```suggestion
The extra spec `scs:name-vN` is to be interpreted as "name variant N". This name scheme is designed to be
```

## Conformance Tests

The script [`flavors-openstack.py`](https://github.com/SovereignCloudStack/standards/blob/main/Tests/iaas/standard-flavors/flavors-openstack.py)
will read the lists of mandatory and recommended flavors
from a yaml file provided as command-line argument, connect to an OpenStack installation,
and check whether the flavors are present and their extra specs are correct.

Missing flavors will be reported on various logging channels: error for mandatory, warning for
recommended flavors. Incorrect extra specs will be reported as error in any case.
The return code will be non-zero if the test could not be performed or if any error was
reported.

> The script does not check whether a name given via `scs:name-vN` is indeed valid according

Reviewer: ```suggestion
The script does not check whether a name given via the extra spec `scs:name-vN` is indeed valid according
```
